src/sentry/eventstream/kafka/consumer.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import absolute_import
+
+import logging
+import uuid
+
+from confluent_kafka import Consumer, OFFSET_BEGINNING, TopicPartition
+
+from sentry.eventstream.kafka.state import PartitionState, SynchronizedPartitionStateManager
+from sentry.eventstream.kafka.utils import join
+
+
+logger = logging.getLogger(__name__)
+
+
+class SynchronizedConsumer(object):
+    def __init__(self, bootstrap_servers, topics, consumer_group, commit_log_topic, synchronize_commit_group):
+        self.bootstrap_servers = bootstrap_servers
+        self.topics = topics
+        self.consumer_group = consumer_group
+        self.commit_log_topic = commit_log_topic
+        self.synchronize_commit_group = synchronize_commit_group
+
+    def _commit_callback(self, error, partitions):
+        if error is not None:
+            logger.warning('Failed to commit offsets (error: %s, partitions: %r)', error, partitions)
```

Comment:
This is only a problem if it happens 100% of the time. The worst case scenario here is then the consumer crashes and restarts from an earlier position and re-enqueues/executes some tasks. The task should be updated to prevent this from happening (if required) anyway